fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/typing/Type.java

```diff
@@ -598,7 +607,7 @@ static Type fromObject(@Nullable final Object object) {
             return Type.any();
         }
         if (typeCode.isPrimitive()) {
-            return Type.primitiveType(typeCode, false);
+            return Type.primitiveType(typeCode, isNullable);
```

Comment:
before the change, any Type created by Type.fromObject, will have all primitive types isNullable = false. Add this method to allow callers to set isNullable.

Reply:
We have seen `promote(x as string)` before, it was because, initially x is set to `string, isNullable = false` by this `fromObject` method, and then it was promoted to `string, isNullable = true`.